usecases/scheduled_execution/metrics_collection_job.go

```diff
@@ -113,3 +127,38 @@ func (w MetricCollectionWorker) saveWatermark(ctx context.Context, newWatermarkT
 		return w.repository.SaveWatermark(ctx, tx, nil, models.WatermarkTypeMetrics, nil, newWatermarkTime, nil)
```

Comment:
Do you plan on there having more queries in tha transaction? If not, we can skip it.